integration_tests/tests/eoapi_tests/test_stac_ingestion.py

```diff
@@ -0,0 +1,123 @@
+import time
+
+import pystac
+import pytest
+
+INSERTION_WAIT_TIME = 10
+
+# Test validating the collection
+pytest.mark.order(0)
+
+
+def test_validate_collection(test_collection):
+    pystac.validation.validate_dict(test_collection)
+
+
+# Test validating the item
+pytest.mark.order(1)
+
+
+def test_validate_item(test_item):
+    pystac.validation.validate_dict(test_item)
+
+
+# Test inserting collection
+pytest.mark.order(2)
+
+
+def test_insert_collection(stac_ingestion_instance, test_collection):
+    response = stac_ingestion_instance.insert_collection(test_collection)
+    assert response.status_code in [
+        200,
+        201,
+    ], f"Failed to insert the test_collection :\n{response.text}"
+    # Wait for the collection to be inserted
+    time.sleep(INSERTION_WAIT_TIME)
+
+
+# Test inserting item
+pytest.mark.order(3)
+
+
+def test_insert_item(stac_ingestion_instance, test_item):
+    response = stac_ingestion_instance.insert_item(test_item)
+    assert response.status_code in [
+        200,
+        201,
+    ], f"Failed to insert the test_item :\n{response.text}"
+    # Wait for the item to be inserted
+    time.sleep(INSERTION_WAIT_TIME)
+
+
+# Test querying collection and verifying inserted collection
+pytest.mark.order(4)
+
+
+def test_query_collection(stac_ingestion_instance, test_collection):
+    response = stac_ingestion_instance.query_collection(test_collection["id"])
+    assert response.status_code in [
+        200,
+        201,
+    ], f"Failed to query the test_collection :\n{response.text}"
```

Comment:
This would be a good opportunity to validate the collection -- currently, collections are serialized _without_ `exclude_none` or `exclude_unset`: https://github.com/developmentseed/eoapi-cdk/blob/239b0a9b19090a7da7d8790cd23cc70257431f03/lib/ingestor-api/runtime/src/collection.py#L22

This leads to invalid STAC in the database (e.g. `assets` is often set to `null`). We could catch those problems in these integration tests by validating (e.g. with `pystac.Collection.from_dict(response.json()).validate()`).